applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py

```diff
@@ -60,6 +60,7 @@ def AddVariables(self):
         self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
         self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
         self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_AREA)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_H)
```

Comment:
last comment: If you don't need this to be historical (at least looking at your code I think you don't) then I recommend using it as non-historical which will reduce the memory consumption

Reply:
I am using it as historical for performance reasons, because `GetValue` is slower than `GetSolutionStepValue`, but I don't really need this to be historical.

Reply:
the same with other variables

Reply:
ok makes sense